.github/workflows/wsl-exec.yaml

```diff
@@ -0,0 +1,184 @@
+name: Execute bash scripts in WSL
+
+concurrency: azure-vm
+
+on:
+  workflow_call:
+    secrets:
+      az_creds:
+        description: Credentials to use for Azure
+        required: true
+      az_vm_hostname:
+        description: Hostname for "xfreerdp /v:az_vm_hostname"
+        required: true
+      az_vm_username:
+        description: Username for "xfreerdp /u:az_vm_username"
+        required: true
+      az_vm_password:
+        description: Password for "xfreerdp /p:az_vm_password"
+        required: true
+      certificate:
+        description: Certificate used to enable connecting to Azure
+        required: true
+      certificate_password:
+        description: Pasword to the secret certificate
+        required: true
+    inputs:
+      exec:
+        description: Script to run on WSL
+        required: true
+        type: string
+      repo:
+        description: The url to clone the repo from
+        required: true
+        type: string
+      commit:
+        description: The commit to checkout
+        required: true
+        type: string
+      az_resource_group:
+        description: Name of the Azure resource group to run on
+        required: true
+        type: string
+      az_name:
+        description: Name of the Azure VM to run on
+        required: true
+        type: string
+
+jobs:
+  start-vm:
+    strategy:
+      max-parallel: 1
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Start Azure VM
+        uses: ./.github/actions/setup_vm/
+        with:
+          az_name: ${{ inputs.az_name }}
+          az_resource_group: ${{ inputs.az_resource_group }}
+          az_creds: ${{ secrets.az_creds }}
+  
+  remote-desktop:
+    strategy:
+      max-parallel: 1
+    needs: start-vm
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Connect via RDP
+        uses: ./.github/actions/rdp/
+        with:
+          az_vm_hostname: ${{ secrets.AZURE_VM_AUTHORITY }}
+          az_vm_username: ${{ secrets.AZURE_VM_UN }}
+          az_vm_password: ${{ secrets.AZURE_VM_UP }}
+
+  wsl-setup-and-script-execution:
+    strategy:
+      max-parallel: 1
+    needs: start-vm
+    runs-on: self-hosted
+    env:
+      LAUNCHER: "ubuntupreview.exe"
+      DISTRO: "Ubuntu-Preview"
+      APPX: "Ubuntu (Preview)"
+      WSL_UTF8: 1
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Vet input
+        env:
+          EXEC_TO_VET: ${{ inputs.exec }}
+        shell: powershell
+        run: |
+          If(${EXEC_TO_VET} -match "\n'@" -or ${EXEC_TO_VET} -match "^'@") {
+            Write-Error "Your script contains '@ at the begining of a line. This is not allowed because it would escape the Winows' quotes when invoking it."
```

Comment:
```suggestion
            Write-Error "Your script contains '@ at the begining of a line. This is not allowed because it would escape the Windows' quotes when invoking it."
```